symbolic-symcache/src/new/writer.rs

```diff
@@ -308,6 +313,113 @@ impl SymCacheConverter {
         }
     }
 
+    #[cfg(feature = "il2cpp")]
+    /// Processes a set of [`UsymSymbols`], passing all mapped symbols into the converter.
+    pub fn process_usym(&mut self, usym: &UsymSymbols) -> Result<(), SymCacheError> {
+        // Assume records they are sorted by address; There's a test that guarantees this
+        let mapped_records = usym.records().filter_map(|r| match r {
+            UsymSourceRecord::Unmapped(_) => None,
```

Comment:
This means we are discarding all source records that reference C++?
That in turn would mean the symcache would be useless to symbolicate a C++ crash / minidump?

Reply:
I think that's the idea here, at least for the first step. To create a symcache from the usym that is only usable for managed (C#) code.

Reply:
what @loewenheim said, yeah. this is basically a first step targeting very simple projects that presumably only include c# code. 

a really quick search suggests to me that unity doesn't directly support c++, but there are plugins that could be in c++. would we, or unity devs have access to debug info for those plugins? there's also been a little bit of discussion about whether it would be useful to keep the c++ mappings, but there we didn't really arrive at a concrete conclusion on what we would like to do with those. would slightly mangled symbols for unity engine calls be useful for the end user?

the discarded records here do contain data that's still useful for managed code but they don't come with any managed code data, so we'll probably want to revisit these discarded records at some point.